src/lib.rs

```diff
@@ -342,6 +342,30 @@ impl CspList {
             })
             .next()
     }
+    /// https://www.w3.org/TR/CSP/#can-compile-strings
+    pub fn is_js_evaluation_allowed(&self) -> CheckResult {
+        let mut allowed = CheckResult::Allowed;
+        for policy in &self.0 {
+            for directive in &policy.directive_set {
+                if matches!(allowed, CheckResult::Allowed) {
+                    allowed = directive.is_js_evaluation_allowed(&policy);
```

Comment:
Nit: I can see that this is written to match the spec, but an early return on `AllowResult::DoesNotAllow` seems both faster and more clear. Looking at `is_base_allowed_for_document` above, we do use early returns like that.

But that's my non-binding opinion, I'm not closely involved with the development of this library.